controllers/chat/sendAnonymousMessage.js

```diff
@@ -101,7 +109,42 @@ const sendAnonymousMessage = async (req, res) => {
       });
     }
 
-    const chat = await channel.sendMessage(baseMessage);
+    const chat = await channel.sendMessage(baseMessage, {skip_push: true});
+    let notificationPayload = {
+      title: `New Message from ${senderInfo?.user?.username}`,
+      body: `${message.substring(0, 100)}`
+    };
+    let dataPayload = {
+      channel_id: channelId,
+      user_id: req.userId,
+      message_id: chat.message.id,
+      message,
+      message_schema: currentMessageType,
+      type: 'message.new',
+      status: 'sent',
+      is_big_message: 'false',
+      attachment: JSON.stringify(attachments) ?? '',
+      created_at: chat.message.created_at,
+      is_annoymous: 'false'
+    };
+    const dataSizeInBytes = Buffer.byteLength(JSON.stringify(dataPayload), 'utf8');
+    const dataSizeInKilobytes = dataSizeInBytes / 1024;
+    if (dataSizeInKilobytes > 4) {
+      dataPayload.is_big_message = 'true';
+      dataPayload.attachment = '';
+    }
+    channelMember.forEach((member) => {
+      let payload = {
+        notification: notificationPayload,
+        // priority: 'high',
+        // content_available: true,
+        data: {
+          ...dataPayload,
+          is_annoymous: member.is_anonymous.toString()
+        }
+      };
+      BetterSocialCore.fcmToken.sendChatNotification(member.user_id, payload);
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code for creating the notification payload and sending it to each channel member is quite complex and could be simplified. Consider extracting this logic into a separate function to improve readability and maintainability. Also, there's no error handling around `BetterSocialCore.fcmToken.sendChatNotification`. If an error occurs during the notification sending process, it will crash the entire application. Consider adding error handling.
```diff
+   const createNotificationPayload = (message, senderInfo, currentMessageType, attachments) => {
+     let notificationPayload = {
+       title: `New Message from ${senderInfo?.user?.username}`,
+       body: `${message.substring(0, 100)}`
+     };
+     let dataPayload = {
+       channel_id: channelId,
+       user_id: req.userId,
+       message_id: chat.message.id,
+       message,
+       message_schema: currentMessageType,
+       type: 'message.new',
+       status: 'sent',
+       is_big_message: 'false',
+       attachment: JSON.stringify(attachments) ?? '',
+       created_at: chat.message.created_at,
+       is_annoymous: 'false'
+     };
+     const dataSizeInBytes = Buffer.byteLength(JSON.stringify(dataPayload), 'utf8');
+     const dataSizeInKilobytes = dataSizeInBytes / 1024;
+     if (dataSizeInKilobytes > 4) {
+       dataPayload.is_big_message = 'true';
+       dataPayload.attachment = '';
+     }
+     return {notificationPayload, dataPayload};
+   }

+   const sendNotifications = async (channelMembers, notificationPayload, dataPayload) => {
+     for (let member of channelMembers) {
+       let payload = {
+         notification: notificationPayload,
+         data: {
+           ...dataPayload,
+           is_annoymous: member.is_anonymous.toString()
+         }
+       };
+       try {
+         await BetterSocialCore.fcmToken.sendChatNotification(member.user_id, payload);
+       } catch (error) {
+         console.error(`Failed to send notification to user ${member.user_id}:`, error);
+       }
+     }
+   }

-   const chat = await channel.sendMessage(baseMessage, {skip_push: true});
-   let notificationPayload = {
-     title: `New Message from ${senderInfo?.user?.username}`,
-     body: `${message.substring(0, 100)}`
-   };
-   let dataPayload = {
-     channel_id: channelId,
-     user_id: req.userId,
-     message_id: chat.message.id,
-     message,
-     message_schema: currentMessageType,
-     type: 'message.new',
-     status: 'sent',
-     is_big_message: 'false',
-     attachment: JSON.stringify(attachments) ?? '',
-     created_at: chat.message.created_at,
-     is_annoymous: 'false'
-   };
-   const dataSizeInBytes = Buffer.byteLength(JSON.stringify(dataPayload), 'utf8');
-   const dataSizeInKilobytes = dataSizeInBytes / 1024;
-   if (dataSizeInKilobytes > 4) {
-     dataPayload.is_big_message = 'true';
-     dataPayload.attachment = '';
-   }
-   channelMember.forEach((member) => {
-     let payload = {
-       notification: notificationPayload,
-       // priority: 'high',
-       // content_available: true,
-       data: {
-         ...dataPayload,
-         is_annoymous: member.is_anonymous.toString()
-       }
-     };
-     BetterSocialCore.fcmToken.sendChatNotification(member.user_id, payload);
-   });

+   const chat = await channel.sendMessage(baseMessage, {skip_push: true});
+   const {notificationPayload, dataPayload} = createNotificationPayload(message, senderInfo, currentMessageType, attachments);
+   await sendNotifications(channelMembers, notificationPayload, dataPayload);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->